org.lflang/src/org/lflang/federated/generator/FedTargetConfig.java

```diff
@@ -57,11 +58,20 @@ private void mergeImportedConfig(
       if (targetProperties != null) {
         // Merge properties
         TargetProperty.update(
-            this, convertToEmptyListIfNull(targetProperties.getPairs()), errorReporter);
+            this,
+            convertToEmptyListIfNull(targetProperties.getPairs()),
+            getRelativePath(mainResource, federateResource),
+            errorReporter);
       }
     }
   }
 
+  private Path getRelativePath(Resource source, Resource target) {
```

Comment:
Isn't there a method in `FedFileConfig` that does something like this?

Reply:
Yes, it is similar, and it is better in that it handles important cases that I did not address here. However, it produces paths that are relative to the generated LF file, whereas here we need paths relative to the original main LF file.

The "right way" to fix this is to require the projects consisting of multiple files to specify target properties in a separate Lingo config file. I would be OK with merging this as a stopgap measure until we can do it the right way.

Reply:
Sounds fair to me...